junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -106,7 +112,7 @@ inline fun <reified T : Throwable> assertThrows(noinline executable: () -> Unit)
  * ```
  * @see Assertions.assertThrows
  */
-inline fun <reified T : Throwable> assertThrows(message: String, noinline executable: () -> Unit): T =
+inline fun <reified T : Throwable> assertThrows(message: String, noinline executable: suspend () -> Unit): T =
```

Comment:
Is this backwards compatible?

Reply:
Non-suspending functions can be passed here as well, if this is what you mean with backwards compatible.

Reply:
More from a binary standpoint. Will it change the bytecode signature of the method.
I don't think that that's a problem because these are `inline` methods so they aren't actually called.